CMSgov/onemac
# OY2-16335: Add Package Actions for Base Waivers and Renewal Waivers: Temporary Extension

Story: https://qmacbis.atlassian.net/browse/OY2-16335
Endpoint: https://d1cs35nowomkub.cloudfront.net/

### Details
Add the "Request a Temporary Extension" action to the Action Popups and Detail page Links

### Changes
- moved ConfirmationDialog to be able to be used through the appContext (leaving it as-is in the PROD areas)
- updated Package Pages to use context confirmations
- pulled Action management out of containing pages, combined with the "popup" code and created ActionPopup
- added "Request Temporary Extension" as a package action that goes to the temporary extension page with the caller's id listed as the parent.

### Implementation Notes
Fixed a bug in the v0 versioning - the v1's and v2's were missing some of the fields.
Clean up - forms do not use actionsByStatus, so removed that from their configs.
Attempted to consolidate Actions such that one process handles the details about the Action, regardless of whether it is called by ActionPopup or the Package Action Links - not there yet :(  

### Test Plan
1. Login as a state submitter
2. Navigate to Package Dashboard
3. Click to Waivers tab
4. Locate a Waiver in "Approved" status
5. Click Action kebab, verify "Request Temporary Extension" is an available Action
6. Click "Request Temporary Extension" and verify it goes to the Temporary Extension Form with the correct Parent indicated
7. Return to Package Dashboard
8. Locate Waiver in "Approved" status
9. click Waiver number link to go into the Waiver Details
10. Verify "Request a Temporary Extension" is an available Package Action Link
11. Click "Request a Temporary Extension" and verify it goes to the Temporary Extension Form with the correct Parent indicated
